### PR TITLE
Add generateEmbeddedObjectMeta flag when generating crd

### PR DESCRIPTION
**Pull request by SpiritZhou**
Add generateEmbeddedObjectMeta flag when generating crd

### Checklist

- [ ] When introducing a new scaler, I agree with the [scaling governance policy](https://github.com/kedacore/governance/blob/main/SCALERS.md)
- [ ] I have verified that my change is according to the [deprecations & breaking changes policy](https://github.com/kedacore/governance/blob/main/DEPRECATIONS.md)
- [ ] Tests have been added
- [X] Changelog has been updated and is aligned with our [changelog requirements](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#Changelog)
- [ ] A PR is opened to update our Helm chart ([repo](https://github.com/kedacore/charts)) *(if applicable, ie. when deployment manifests are modified)*
- [ ] A PR is opened to update the documentation on ([repo](https://github.com/kedacore/keda-docs)) *(if applicable)*
- [X] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://github.com/kedacore/keda/blob/main/CONTRIBUTING.md#developer-certificate-of-origin-signing-your-work))

Relates to # https://github.com/kedacore/keda/issues/5908

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -60,6 +60,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 - TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
 - **General**: Add --ca-dir flag to KEDA operator to specify directories with CA certificates for scalers to authenticate TLS connections (defaults to /custom/ca) ([#5860](https://github.com/kedacore/keda/issues/5860))
+- **General**: Add the generateEmbeddedObjectMeta flag when generating CRD ([#5908](https://github.com/kedacore/keda/issues/5908))
```
Comment (tomkerkhove) [inline CHANGELOG.md:63]: We may want to elaborate on what this means for end-users though? Why does this matter?

File: CHANGELOG.md [hunk 2]
```diff
@@ -60,6 +60,7 @@ To learn more about active deprecations, we recommend checking [GitHub Discussio
 
 - TODO ([#XXX](https://github.com/kedacore/keda/issues/XXX))
 - **General**: Add --ca-dir flag to KEDA operator to specify directories with CA certificates for scalers to authenticate TLS connections (defaults to /custom/ca) ([#5860](https://github.com/kedacore/keda/issues/5860))
+- **General**: Add the generateEmbeddedObjectMeta flag to generate meta properties of JobTargetRef in ScaledJob ([#5908](https://github.com/kedacore/keda/issues/5908))
```
Comment (JorTurFer) [inline CHANGELOG.md:63]: This will need to be moved to the current version

**Comment by JorTurFer** [comment]
LGTM and it's a nice fix indeed!

**Review by JorTurFer — APPROVED** [review]

**Pull request merged**